### PR TITLE
[MOSIP-23969]

### PR DESCRIPTION
**Pull request by bhumi46**
added preinstall job for openssl creation (regclient)

### CODE REVIEW

**Review by syedsalman3753 — APPROVED** [review]

File: charts/regclient/templates/pre-install-job.yaml
Comment (ckm007) [inline charts/regclient/templates/pre-install-job.yaml:1]: keystore-job.yaml

File: charts/regclient/templates/pre-install-job.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "{{.Release.Name}}-{{ .Release.Namespace }}-job"
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/managed-by: {{.Release.Service | quote }}
+    app.kubernetes.io/instance: {{.Release.Name | quote }}
+    helm.sh/chart: "{{.Chart.Name}}-{{.Chart.Version}}"
+    app: generate-certs
```
Comment (ckm007) [inline charts/regclient/templates/pre-install-job.yaml:10]: check if needed. if needed make it same as keystore

File: charts/regclient/templates/pre-install-job.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-2"
+    "helm.sh/hook-delete-policy": hook-succeeded
```
Comment (ckm007) [inline charts/regclient/templates/pre-install-job.yaml:14]: dont need delete policy

File: charts/regclient/templates/role-preinstalljob.yaml
Comment (ckm007) [inline charts/regclient/templates/role-preinstalljob.yaml:1]: keystore-role.yaml

File: charts/regclient/templates/role-preinstalljob.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  namespace: {{ .Release.Namespace }}
+  name: {{.Release.Name}}-{{ .Release.Namespace }}-role
```
Comment (ckm007) [inline charts/regclient/templates/role-preinstalljob.yaml:5]: {{.Values.keystore.Name}}-{{ .Release.Namespace }}

File: charts/regclient/templates/role-preinstalljob.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  annotations:
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-weight": "-4"
+rules:
+  - apiGroups: [""]
+    resources: ["pods","configmaps", "secrets"]
+    verbs: ["get","list","create", "update", "delete", "patch"]
```
Comment (ckm007) [inline charts/regclient/templates/role-preinstalljob.yaml:12]: add necesary roles only

**Review by ckm007 — APPROVED** [review]

**Pull request merged**